### ISSUE TITLE
Import maemo-statusmenu-fmtx?

### ISSUE THREAD

**Issue by MerlijnWajer** [maemo-leste/bugtracker#40]
https://github.com/community-ssu/maemo-statusmenu-fmtx

**Comment by MerlijnWajer** [comment]
There is also http://maemo.org/packages/view/maemo-applet-fmtx/ (control panel)

**Comment by freemangordon** [comment]
this is closed source :)
what we need is https://github.com/community-ssu/maemo-statusmenu-fmtx and https://github.com/community-ssu/fmtx-middleware

**Comment by MerlijnWajer** [comment]
So Alsa UCM can toggle v4l keys if we want to, which makes me think that with a simple pulseaudio output change we can enable FM and disable it too. Not sure if that makes the most sense, but FYI.

**Comment by MerlijnWajer** [comment]
And I assumed the middleware would do v4l2, but I don't think it does, right?

**Comment by MerlijnWajer** [comment]
>this is closed source :)
oops!

**Comment by sicelo** [comment]
On Wed, Jun 14, 2023 at 10:18:35AM -0700, Merlijn Wajer wrote:
> So Alsa UCM can toggle v4l keys if we want to, which makes me think that with a simple pulseaudio output change we can enable FM and disable it too. Not sure if that makes the most sense, but FYI.

UCM/alsa only handles the audio routing. The FM transmitter chip must
still be enabled and configured (by either v4l, or you some other way).

Looks like `fmtx-middleware` does require something of v4l ... the
inclusion of `linux/videodev.h`

**Comment by MerlijnWajer** [comment]
>UCM/alsa only handles the audio routing. The FM transmitter chip must
still be enabled and configured (by either v4l, or some other way).

You can execute arbitrary commands via UCM, so I think you can call v4l2-ctl that way

**Comment by freemangordon** [comment]
we need status menu plugin (https://github.com/community-ssu/maemo-statusmenu-fmtx) that sets the frequency and tells fmtxd (https://github.com/community-ssu/fmtx-middleware) to enable the chip. Not sure about audio routing though - is there anything that shall be done at all?

**Comment by freemangordon** [comment]
it seems fmtxd already talks to PA - https://github.com/community-ssu/fmtx-middleware/blob/master/audio.c#L271

**Comment by MerlijnWajer** [comment]
Yes, alsa mixer controls also need to be set and with our new UCM we now have the ability to properly/easily toggle based on use case.

This middleware seems to write to sysfs, isn't it supposed to use v4l controls? (here https://github.com/community-ssu/fmtx-middleware/blob/master/audio.c#L202)

**Comment by freemangordon** [comment]
well, obviously it needs porting :). but this is fremantle fmtxd RE-ed
